CRM/Civicase/Service/CaseTypeCategoryEventHandler.php

```diff
@@ -36,7 +36,7 @@ class CRM_Civicase_Service_CaseTypeCategoryEventHandler {
    * @param \CRM_Civicase_Service_CaseCategoryCustomFieldExtends $customFieldExtends
    *   Custom field handler.
    */
-  public function __construct($menu, $customData, $customFieldExtends) {
+  public function __construct(\CRM_Civicase_Service_CaseCategoryMenu $menu, \CRM_Civicase_Service_CaseCategoryCustomDataType $customData, \CRM_Civicase_Service_CaseCategoryCustomFieldExtends $customFieldExtends) {
```

Comment:
The type hinting makes this line very long. 
How about we alias this class at the use them here.

```php
use CRM_Civicase_Service_CaseCategoryMenu as CaseCategoryMenu
use CRM_Civicase_Service_CaseCategoryCustomDataType as CaseCategoryCustomDataType
use CRM_Civicase_Service_CaseCategoryCustomFieldExtends as CaseCategoryCustomFieldExtends

public function __construct(CaseCategoryMenu $menu, CaseCategoryCustomDataType $customData, CaseCategoryCustomFieldExtends $customFieldExtends) 
```